.github/workflows/thread-sanitizer.yaml

```
@@ -27,139 +31,106 @@ on:
-
-      - name: Build & install Fast-CDR
-        run: |
-          cd ws
-          cmake -DCMAKE_INSTALL_PREFIX="./install" -DCMAKE_PREFIX_PATH="./install" `
-                -DCMAKE_MODULE_PATH="$(pwd)/sanitizers-cmake/cmake" -DCMAKE_VERBOSE_MAKEFILE=ON `
-                -DSANITIZE_THREAD=ON -DCMAKE_BUILD_TYPE=Debug -B ./build/fastcdr ./Fast-CDR
-          cmake --build ./build/fastcdr --target install --verbose
+      - uses: eProsima/eProsima-CI/ubuntu/install_python_packages@v0
+        with:
+          packages: vcstool
 
-      - name: Download foonathan/memory and make it sanitizer aware
-        run: |
-          cd ws
-          git clone https://github.com/foonathan/memory.git
-          cd memory
-          git remote add efork https://github.com/MiguelBarro/memory.git
-          git fetch efork
-          git merge --no-edit efork/bugfix/add_sanitizer_support
-
-      - name: Build & install foonathan/memory
-        run: |
-          cd ws
-          cmake -DCMAKE_MODULE_PATH="$(pwd)/sanitizers-cmake/cmake" -DCMAKE_VERBOSE_MAKEFILE=ON -DSANITIZE_THREAD=ON `
-                -DCMAKE_INSTALL_PREFIX="./install" -DFOONATHAN_MEMORY_BUILD_TOOLS=ON -DCMAKE_POSITION_INDEPENDENT_CODE=ON `
-                -DFOONATHAN_MEMORY_BUILD_TESTS=ON -DFOONATHAN_MEMORY_BUILD_EXAMPLES=ON -DFOONATHAN_MEMORY_BUILD_TOOLS=ON  `
-                -DCMAKE_BUILD_TYPE=Debug -B ./build/memory ./memory
-          cmake --build ./build/memory --target install --verbose
-
-      - name: Download Fast-DDS and make it sanitizer aware
+      - uses: eProsima/eProsima-CI/ubuntu/get_file_from_repo@v0
+        with:
+          source_repository: eProsima/Fast-DDS
+          source_repository_branch: ${{ env.FASTDDS_BRANCH }}
+          file_name: fastrtps.repos
+          file_result: fastrtps.repos
+
+      - name: Fetch Fast DDS & dependencies
+        uses: eProsima/eProsima-CI/ubuntu/vcs_import@v0
+        with:
+          vcs_repos_file: fastrtps.repos
+          destination_workspace: src
+
+      - name: Checkout Fast DDS branch
         run: |
-          cd ws
-          git clone https://github.com/eProsima/Fast-DDS.git
-          cd Fast-DDS
-          git fetch origin ${{ github.ref }}:merge-branch
-          git checkout merge-branch
-          echo "Modified files against destination branch"
-          git diff --name-only HEAD~1
-          git remote add efork https://github.com/MiguelBarro/Fast-DDS.git
-          git fetch efork
-          git merge --no-edit efork/bugfix/add_sanitizer_support
-
-      - name: Build & install Fast-DDS
+          cd ./src/fastrtps
+          git checkout ${{ env.FASTDDS_BRANCH }}
+
+      - name: Install apt packages
+        uses: ./src/fastrtps/.github/actions/install-apt-packages
+
+      - name: Install Python packages
+        uses: ./src/fastrtps/.github/actions/install-python-packages
+
+      - name: Install GTest
+        uses: eProsima/eProsima-CI/ubuntu/install_gtest@v0
+        with:
+          version: v1.12.1
```

This is what we are using elsewhere (see [here](https://fast-dds.docs.eprosima.com/en/latest/installation/sources/sources_linux.html#gtest))
```suggestion
          version: release-1.11.0
```